flink-clients/src/main/java/org/apache/flink/client/CliFrontend.java

```diff
@@ -132,9 +132,9 @@
 
 
 
-	private final Configuration config;
+	private Configuration config;
```

Comment:
Could we make the `configDir` parameter a general parameter that goes before `run`, `list`, `cancel`, etc.? Don't know if that is too tricky but it would make sense because it is a general configuration key.

`flink --configDir /my/custom/dir run -p 10 /path/to/jar`

That way the config parameter wouldn't have to be listed for every action. Would make the help look less verbose.

You could parse the config parameter in the constructor and initialize the config only once. Then continue with the normal parsing.